src/_pytest/deprecated.py

```diff
@@ -95,6 +95,8 @@
     "see https://docs.pytest.org/en/latest/deprecations.html#node-fspath-in-favor-of-pathlib-and-node-path",
 )
 
+MARKED_FIXTURE = PytestDeprecationWarning("Marks cannot be applied to fixtures")
```

Comment:
Maybe make this message more informative, e.g. "Marks applied to fixture functions have no effect. Use XYZ instead."

Reply:
there's not always a clean option, eg sometimes you want `item.add_marker(` sometimes you want to use `@pytest.fixture(params=[pytest.param(..., marks=...)])` and sometimes you want to add an explicit fixture dependency

Reply:
Right, this is probably too much for the warning itself, but maybe in the deprecation doc, basically what you wrote above would be helpful I think. But OK if not.